cpp/tensorrt_llm/thop/groupRmsNormOp.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2022-2025, NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "tensorrt_llm/common/cudaUtils.h"
+#include "tensorrt_llm/common/dataType.h"
+#include "tensorrt_llm/common/opUtils.h"
+#include "tensorrt_llm/kernels/groupRmsNormKernels/groupRmsNormKernels.h"
+#include "tensorrt_llm/runtime/torchUtils.h"
+#include "tensorrt_llm/thop/thUtils.h"
+
+#include <ATen/cuda/CUDAContext.h>
+#include <cstdint>
+#include <cuda_runtime.h>
+#include <stdexcept>
+#include <torch/extension.h>
+#include <vector>
+
+namespace torch_ext
+{
+
+void groupRMSNorm(torch::TensorList const& inputs, torch::TensorList const& outputs, torch::TensorList const& weights,
```

Comment:
How about fusing these two ops into one and determine using `gropuRMSNorm` or `groupRMSNormLargeBatch` depending on the `inputs.shape[0]`?

Reply:
Yes, that will be added in the next PR.